### ISSUE TITLE
Audit usage of explicit `aria-label` on Button component implementations

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#66271]
### Description

It appears that in some cases some `Button` components use an explicit `aria-label` prop to render an `aria-label` HTML attribute, which in turn determines the icon button tooltip (if it's an icon button). Example:

https://github.com/WordPress/gutenberg/blob/91272a2b4bfca1f75a6754a0f3689c3a42988f0c/packages/block-editor/src/components/border-radius-control/linked-button.js#L12-L21

This pattern should be audited across the entire codebase as it's not a best practice. 

Setting an explicit `aria-label` signals two possible cases, both less than ideal:
- The `aria-label` is used in a way that the accessible name mismatches the Button _text_ or tooltip. This is not OK for accessibility.
- The `aria-label` is just unnecessary, like in the example above. In that case, the wrapping Tooltip component uses the same `label` variable  used for `aria-label`. In this case, the Tooltip component can be removed and the variable passed to the `label` prop of the Butotn.

### Step-by-step reproduction instructions

N/A

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

**Comment by ciampo** [comment]
Related to https://github.com/WordPress/gutenberg/pull/66021

cc @WordPress/gutenberg-components

**Comment by im3dabasia** [comment]
Hey @afercia , 

I have studied the issue you described above and found three instances that follow a similar pattern. I have attached my findings below. 

Please provide me with feedback on them, and if these need to be fixed, may I open PRs for the same?

https://github.com/WordPress/gutenberg/blob/114aa686fa86a59e37d83f896551f59f74080709/packages/block-editor/src/components/spacing-sizes-control/linked-button.js#L12-L20

https://github.com/WordPress/gutenberg/blob/114aa686fa86a59e37d83f896551f59f74080709/packages/components/src/box-control/linked-button.tsx#L20-L29

https://github.com/WordPress/gutenberg/blob/114aa686fa86a59e37d83f896551f59f74080709/packages/components/src/border-box-control/border-box-control-linked-button/component.tsx#L28-L39

**Edit: I'll Raise a PR to fix the above changes soon**
